clang/test/CodeGenSYCL/simple-sycl-virtual-function.cpp

```diff
@@ -0,0 +1,68 @@
+// REQUIRES: CPU
+// Adjust when uninitialized global variables are supported on GPU
+
+// RUN: clang++ -cc1 -fsycl-allow-virtual-functions -fsycl-is-device -internal-isystem Inputs -emit-llvm %s -o - | FileCheck %s
+// RUN: clang++ -cc1 -fsycl-is-device -internal-isystem Inputs -emit-llvm %s -o - &> %t.txt | FileCheck %s --input-file %t.txt --check-prefix CHECK-WO-KEY
+
+// CHECK: @_ZTVN10__cxxabiv120__si_class_type_infoE = external addrspace(1) global ptr addrspace(1)
+// CHECK: @_ZTVN10__cxxabiv117__class_type_infoE = external addrspace(1) global ptr addrspace(1)
+// CHECK: @_ZTI4Base = linkonce_odr constant { ptr addrspace(1), ptr } { ptr addrspace(1) getelementptr inbounds (ptr addrspace(1), ptr addrspace(1) @_ZTVN10__cxxabiv117__class_type_infoE, i64 2)
+// CHECK: @_ZTI8Derived1 = linkonce_odr constant { ptr addrspace(1), ptr, ptr } { ptr addrspace(1) getelementptr inbounds (ptr addrspace(1), ptr addrspace(1) @_ZTVN10__cxxabiv120__si_class_type_infoE, i64 2)
+// CHECK: @_ZTI8Derived2 = linkonce_odr constant { ptr addrspace(1), ptr, ptr } { ptr addrspace(1) getelementptr inbounds (ptr addrspace(1), ptr addrspace(1) @_ZTVN10__cxxabiv120__si_class_type_infoE, i64 2)
+
+// CHECK-WO-KEY: SYCL kernel cannot call a virtual function
```

Comment:
Lack or presence of a diagnostic should be checked in SemaSYCL test through `-verify`. CodeGen test should only be concerned about LLVM IR we emit

Reply:
Ok 8a49409